libbeat/conditions/network.go

```diff
@@ -157,15 +157,23 @@ func (c *Network) Check(event ValuesMap) bool {
 			return false
 		}
 
-		ip := extractIP(value)
-		if ip == nil {
+		ipList := extractIP(value)
+		if len(ipList) == 0 {
 			c.log.Debugf("Invalid IP address in field=%v for network condition", field)
 			return false
 		}
-
-		if !network.Contains(ip) {
+		// match on an "any" basis when we find multiple IPs in the event;
+		// if the network matcher returns true for any seen IP, consider it a match
+		matches := 0
+		for _, ip := range ipList {
+			if network.Contains(ip) {
+				matches += 1
```

Comment:
I think you can remove `matches` by using a `continue` here and an unconditional `return false` after the loop.